sound/soc/sof/sof-audio.c

```diff
@@ -419,6 +552,33 @@ snd_sof_find_swidget_sname(struct snd_soc_component *scomp,
 	return NULL;
 }
 
+/* find widget by name */
+struct snd_sof_widget *snd_sof_find_swidget_by_name(struct snd_soc_component *scomp,
+						    const char *name)
+{
+	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(scomp);
+	struct snd_sof_widget *swidget;
+
+	list_for_each_entry(swidget, &sdev->widget_list, list)
+		if (!strcmp(name, swidget->widget->name))
+			return swidget;
+
+	return NULL;
+}
+
+/* find widget by comp_id */
+struct snd_sof_widget *snd_sof_find_swidget_by_id(struct snd_soc_component *scomp, int id)
+{
+	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(scomp);
+	struct snd_sof_widget *swidget;
+
+	list_for_each_entry(swidget, &sdev->widget_list, list)
+		if (swidget->comp_id == id)
+			return swidget;
+
+	return NULL;
+}
```

Comment:
as long as we don't have external users for these functions, I'd keep them static. Keeps symbol-tables small :-)